src/HEVM.hs

```diff
@@ -1,446 +1,365 @@
-  UIntMin _ a -> show $ uintmin a
-  UIntMax _ a -> show $ uintmax a
-  IntEnv _ a -> nameFromEnv c m a
-
-  And _ a b   -> nameFromExp c m a <> "&&" <> nameFromExp c m b
-  Or _ a b    -> nameFromExp c m a <> "|" <> nameFromExp c m b
-  Impl _ a b  -> nameFromExp c m a <> "=>" <> nameFromExp c m b
-  LT _ a b    -> nameFromExp c m a <> "<" <> nameFromExp c m b
-  LEQ _ a b   -> nameFromExp c m a <> "<=" <> nameFromExp c m b
-  GT _ a b    -> nameFromExp c m a <> ">" <> nameFromExp c m b
-  GEQ _ a b   -> nameFromExp c m a <> ">=" <> nameFromExp c m b
-  Neg _ a     -> "~" <> nameFromExp c m a
-  LitBool _ a -> show a
-  Eq _ _ a b    -> nameFromExp c m a <> "=="  <> nameFromExp c m b
-  NEq _ _ a b   -> nameFromExp c m a <> "=/=" <> nameFromExp c m b
-  Cat _ a b -> nameFromExp c m a <> "++" <> nameFromExp c m b
-  ByStr _ a -> show a
-  ByLit _ a -> show a
-  Slice _ a x y -> nameFromExp c m a <> "[" <> show x <> ":" <> show y <> "]"
-  ByEnv _ a -> nameFromEnv c m a
-
-  Var _ _ a -> nameFromArg c m a
-  TEntry _ _ a -> nameFromItem c m a
-  ITE _ x y z -> "if-" <> nameFromExp c m x <> "-then-" <> nameFromExp c m y <> "-else-" <> nameFromExp c m z
-
-  Create _ _ _ _ -> error "calls not supported"
-
-nameFromDecl :: ContractName -> Method -> Decl -> Id
-nameFromDecl c m (Decl _ name) = nameFromArg c m name
-
-nameFromArg :: ContractName -> Method -> Id -> Id
-nameFromArg c method name = c @@ method @@ name
-
-nameFromEnv :: ContractName -> Method -> EthEnv -> Id
-nameFromEnv c method e = c @@ method @@ (prettyEnv e)
-
-(@@) :: (Show a, Show b) => a -> b -> Id
-x @@ y = show x <> "_" <> show y
-
--- *** Utils *** --
-
-
-get :: (Show a, Ord a, Show b) => a -> Map a b -> b
-get name vars = fromMaybe (error (show name <> " not found in " <> show vars)) $ Map.lookup name vars
-
-catInts :: Map Id SActType -> Map Id (SBV Integer)
-catInts m = Map.fromList [(name, i) | (name, SymInteger i) <- Map.toList m]
-
-catBools :: Map Id SActType -> Map Id (SBV Bool)
-catBools m = Map.fromList [(name, i) | (name, SymBool i) <- Map.toList m]
-
-catBytes :: Map Id SActType -> Map Id (SBV String)
-catBytes m = Map.fromList [(name, i) | (name, SymBytes i) <- Map.toList m]
-
-concatMapM :: Monad m => (a -> m [b]) -> [a] -> m [b]
-concatMapM f xs = fmap concat (mapM f xs)
+    aux :: Types.Expr Types.End -> Types.Prop
+    aux (Types.Success c _ _) = Types.pand c
+    aux _ = error "List should only contain success behaviours"
+
+-- | Check whether two lists of behaviours cover exactly the same input space
+checkInputSpaces :: SolverGroup -> VeriOpts -> [Types.Expr Types.End] -> [Types.Expr Types.End] -> IO [EquivResult]
+checkInputSpaces solvers _ l1 l2 = do
+  let p1 = inputSpace l1
+  let p2 = inputSpace l2
+  let queries = fmap assertProps [ [ Types.PNeg (Types.por p1), Types.por p2 ]
+                               , [ Types.por p1, Types.PNeg (Types.por p2) ] ]
+
+  when True $ forM_ (zip [(1 :: Int)..] queries) $ \(idx, q) -> do
```

Comment:
`when True`?

Reply:
yes it is a placefolder for the debug flag. The problem is that field selectors are not being exported.

Reply:
ah, we need to enable `OverloadedRecordDot` or smth in act too I guess to be able to get this nice modern dot notation for accessing record fields.

Reply:
On nice! I did it locally for the file and now `opts.debug` works fine.